Test login

cealicea171 opened the pull request:
1- Login and Logout
2- Test for student and teacher.

Commit:
worked on class Auth for confestt.py, added auth fixture and created a test_login test, 5 passing  test

Commit:
worked on login feature, before worked on test was passing, now they arent.

Commit:
got all 6 test passing but poor tests, need to fix.

Commit:
got the login and logout route to work , also added a logout button, test are passing but test need to be improved.

Commit:
finished login and logout features in init
finished 7 passing test with teacher and student log in.

zachfedor commented on portal/__init__.py:
```diff
@@ -18,9 +18,54 @@ def create_app(test_config=None):
+                #WHERE
+                # return redirect(url_for('index'))
+
+            print(error)
```
Lots of `print()` statements, these need cleaned up.

zachfedor commented on tests/conftest.py:
```diff
@@ -35,4 +35,29 @@ def client(app):
+        return self._client.post(
+            '/',
+            data={'email': email, 'password': password}
+        )
```
replace this with:

```python
def login(self, email, password):
    return self._client.post('/', data={'email': email, 'password': password})

def login_teacher(self):
    return self.login('teacher@stevenscollege.edu', 'qwerty')

def login_student(self):
    return self.login('student@stevenscollege.edu', 'asdfgh')
```

zachfedor commented on tests/test_app.py:
```diff
@@ -8,5 +8,4 @@ def test_config():
     assert b'<h1>TSCT Portal</h1>' in response.data
-    assert b'<form>' in response.data
-
+    #assert b'<form>' in response.data
```
clean this up

zachfedor commented on portal/__init__.py:
```diff
@@ -18,9 +18,54 @@ def create_app(test_config=None):
+                print(user[0])
+                print(user[1])
+                session['user_id'] = user[0]
+                session['user_email'] = user[1]
```
instead of setting user info on session cookie, just set user id and use a hook like `@bp.before_app_request` to look at session data and then populate the global `g` object. more info [here](http://flask.pocoo.org/docs/1.0/tutorial/views/)

Commit:
Cleaned up code, got rid of irrelevant comments and print statements

Commit:
-finish login
-finish log out
-complete a log in and out test

Commit:
clean up print statements and comments.

Commit:
Add Tests
-wrote test to check incorrect email and password.
-wrote test check add_user function
